### PR TITLE
docs: contributing code and design new content

### LINKED ISSUE #3719: Website docs: Contributing code page
### Acceptance criteria

- Finalizing Aubrey's [prototype](https://www.figma.com/file/9pdTqNPVJCmDOAacm3Rnxp/Contribution?type=design&node-id=821-102895&mode=design&t=hKf6VqbaJJheia8x-0).
- Ensuring that the "setting up your development env section" in GH is accurate and up-to-date. Cross-link to it.
- Sync with the bot work and ensure the labels are correct.

### PR DESCRIPTION
**Pull request by aubrey-oneal**
Closes #3719 

#### Changelog

**New screen capture videos**
- code-contribution.mp4
- design-contribution.mp4
- edit-this-page-contribution.mp4
- indline-edit.mp4

**New pages**

- Code
- Design

**Pages Removed**

- Bugs and request
- Components
- Patterns
- Icons
- Pictogram
- Add ons

### CODE REVIEW

File: src/pages/contributing/code.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+like to help.
+
+If you want to join an issue that is already assigned to someone or has a
+specific milestone, please discuss with the assignee of the issue or with the
+Carbon team to coordinate.
+
+Issues labeled `good first issue` are great candidates to pick up if you are in
+the code for the first time. The Carbon team is also happy to help you find a
+get started. Just stop by our [office hours](/whats-happening/meetups/) or send
+us a slack message.
```
Comment (andreancardona) [inline src/pages/contributing/code.mdx:48]: Should we add the links to our slack channel here?

**Review by andreancardona — APPROVED** [review]
Small comment / question - otherwise everything looks good!

File: src/pages/contributing/code.mdx (hunk 1 above)
Comment (aubrey-oneal) [reply src/pages/contributing/code.mdx:48]: Good idea! I'll add those.

File: src/pages/contributing/design.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+directly.
+
+#### Submit an idea
+
+Have a new idea that you think would benefit Carbon? Or do you need to report a
+bug? First, be sure to look through the
+[issue backlog](https://github.com/carbon-design-system/carbon/issues) to make
+sure it is a novel idea or bug. Then, file your proposal on GitHub using the
+[issue templates](https://github.com/carbon-design-system/carbon/issues/new/choose).
+If you're willing to work on this idea yourself, be sure to let is know in your
```
Comment (laurenmrice) [inline src/pages/contributing/design.mdx:77]:
```suggestion
If you're willing to work on this idea yourself, be sure to let us know in your
```

**Review by laurenmrice — DISMISSED** [review]
Looking good @aubrey-oneal ! Just found a couple typos:

File: src/pages/contributing/design.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+## Step 3: Iterate and get feedback
+
+Most contributors work in groups of 2-3 and either set up weekly sessions or
+join [meetups](/whats-happening/meetups/#carbon-office-hours) such as the Data
+Viz Guild or office hours. In these sessions, it is common to share work in
+progress and ask lots of questions. As you make progress, update your GitHub
+issue.
+
+Bring any work-in-progress to the following guilds and meet-ups. Here you'll get
```
[the remark this answers is not in the document]
Comment (aubrey-oneal) [reply src/pages/contributing/design.mdx:103]: Good catch! I was going to have a list but decided to limited it to the meetups/contacts within the site. I'll update.

**Review by jeanservaas — CHANGES_REQUESTED** [review]
Looks good, just had a couple of comments to shorten some of the titles and also be specific about which office hours to attend on the code and design tabs.

**Review by jeanservaas — APPROVED** [review]

**Pull request merged**